go/enclave/services/interfaces.go

```diff
@@ -0,0 +1,42 @@
+package services
+
+import (
+	"github.com/obscuronet/go-obscuro/go/common"
+	"github.com/obscuronet/go-obscuro/go/enclave/components"
+	"github.com/obscuronet/go-obscuro/go/enclave/core"
+)
+
+// ObscuroService - the interface for any service type running in Obscuro nodes.
+// Should only contain the shared functionality that every service type needs to have.
+type ObscuroService interface {
+	//	ReceiveBlock - function that accepts L1 blocks and their receipts along with a flag
+	// that signals if the block is the latest one or not. Processing of those blocks and
+	// resulting actions differ between the service types.
+	ReceiveBlock(*common.BlockAndReceipts, bool) (*components.BlockIngestionType, error)
+
+	// SubmitTransaction - L2 obscuro transactions need to be passed here. Sequencers
+	// will put them in the mempool while validators might put them in a queue and monitor
+	// for censorship.
+	SubmitTransaction(*common.L2Tx) error
+}
+
+type Sequencer interface {
+	// CreateBatch - creates a new head batch for the parameter block or if there is no block
+	// provided, for the latest known L1 head block.
+	CreateBatch(*common.L1Block) (*core.Batch, error)
```

Comment:
I'm still contemplating if the host can select what l1 head it wants to create a batch for (and rollup) since it should always know where the l1 chain sits at